src/datatype.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef _DATATYPE_HPP_
+#define _DATATYPE_HPP_
+#include <iostream>
+using ulint = uint64_t;
+using addr_raw_t = uint64_t;
+using addr_t = std::bitset<32>;
+
+inline addr_raw_t Cvt2AddrRaw(addr_t addr) { return addr.to_ullong(); }
+
+inline addr_t Cvt2AddrBits(addr_raw_t raw_addr) {
+    return std::bitset<32>(raw_addr);
+}
+
+enum INST_OP { I_LOAD, I_STORE, I_NONE };
+
+using inst_t = struct inst_s_t {
+    INST_OP op;
+    addr_raw_t addr_raw;
+    inst_s_t() : op(op), addr_raw(addr_raw) {}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Member variable 'addr_raw' is initialized by itself.](https://app.codacy.com/app/jerry0715no14/Cache_sim/pullRequest?prid=4720103)

Reply:
@jackyhuang85  Is there any reason to declare a struct and use `using` to give it another type name?